src/Core/View.js

```diff
@@ -340,7 +341,22 @@ View.prototype.addLayer = function addLayer(layer, parentLayer) {
         }
         layer = _preprocessLayer(this, layer, provider, parentLayer);
         if (parentLayer) {
-            parentLayer.attach(layer);
+            if (layer.type == 'color') {
+                const layerColors = this.getLayers(l => l.type === 'color');
+
+                const sumColorLayers = parentLayer.countColorLayersTextures ?
+                parentLayer.countColorLayersTextures(layerColors, layer) :
+                layerColors.length + 1;
```

Comment:
Why `+1`?

Reply:
by default the number of layers is the same as the number of textures used. (+1 is for added layer)

Reply:
Ah ok.

Some parentheses are needed (`const sumColorLayers = (... ? ... : ...) + 1`) to ease readability and to fix the priority bug in your code: the `+1` only applies in the `layerColors.length` case.

Reply:
there's no bug
by default the number of layers is the same as the number of textures used :  `layerColors.length + 1`

if not, we use `countColorLayersTextures` which returns the number of textures used after adding layer

Reply:
Oh, right. That's because `countColorLayersTextures` counts the not-added layer.

I'd suggest a less confusing API:
```js
wgs84TileLayer.countColorLayersTextures = (...layers) => {
   for (const layer of layers) {
      ...
   }
}
```
And at the calling site use:
```js
parentLayer.countColorLayersTextures(...layerColors, layer)
```

---

As a side note: you're defining this function in GlobeView but it clearly depends on TileMesh and LayeredMaterial so could also live in TileProvider, or in LayeredMaterial... and this is not documented either.

But I don't mind it (= I'm not clicking *Request Changes*) because there's no perfect solution, and this is mainly visible for itowns dev, not itowns user.

(the last paragraph is more a "how to (try to) be a useful reviewer" note, but I felt it was relevant, given the amount of blocked PR we have)